src/web/components/ApiKeyManagement/KeyCreationDialog.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ApiKeySecretsDTO } from '../../../api/services/apiKeyService';
+import { CreateApiKeyFormDTO } from '../../services/apiKeyService';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import KeySecretReveal from '../Core/KeySecretReveal';
+import { StatusPopup } from '../Core/StatusPopup';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { TextInput } from '../Input/TextInput';
+
+import './KeyCreationDialog.scss';
+
+type KeySecretProp = ApiKeySecretsDTO | undefined;
+
+type KeyCreationDialogProps = {
+  onKeyCreation: (form: CreateApiKeyFormDTO) => Promise<KeySecretProp>;
+  triggerButton: JSX.Element;
+  availableRoles: ApiRoleDTO[];
+};
+
+function KeyCreationDialog({
+  onKeyCreation,
+  triggerButton,
+  availableRoles,
+}: KeyCreationDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [secrets, setSecrets] = useState<KeySecretProp>(undefined);
+  const [showStatusPopup, setShowStatusPopup] = useState<boolean>(false);
+  const [statusPopupMessage, setStatusPopupMessage] = useState<string>('');
+  const [copiedSecrets, setCopiedSecrets] = useState<Map<String, boolean>>(
+    new Map<String, boolean>()
+  );
+
+  const onFormSubmit: SubmitHandler<CreateApiKeyFormDTO> = async (formData) => {
+    setSecrets(await onKeyCreation(formData));
+    setStatusPopupMessage(
+      'Your key has been created. Please copy the credentials before closing the window.'
+    );
+    setShowStatusPopup(true);
+  };
+
+  function confirmClose(): void {
+    const values = [...copiedSecrets.values()];
+
+    if (values.filter((value) => value === false).length > 0) {
+      setStatusPopupMessage('Please copy all secrets shown before closing the page');
+      setShowStatusPopup(true);
+      return;
+    }
+
+    setSecrets(undefined);
+    setOpen(false);
+  }
+
+  return (
+    <div className='key-creation-dialog'>
+      <Dialog triggerButton={triggerButton} open={open} onOpenChange={setOpen} hideClose>
+        {!secrets ? (
+          <>
+            <h1>Create API Key</h1>
+            <Form<CreateApiKeyFormDTO> onSubmit={onFormSubmit} submitButtonText='Create API Key'>
+              <TextInput inputName='name' label='Name' required />
+              <CheckboxInput
+                label='API Roles'
+                inputName='roles'
+                options={availableRoles.map((role) => ({
+                  optionLabel: role.externalName,
+                  value: role.roleName,
+                }))}
+                rules={{
+                  required: 'Your new API Key must have at least one role.',
+                }}
+              />
+            </Form>
+            <div className='cancel-container'>
```

Comment:
We can get rid of this div and just use the `closeButton` prop on the `<Dialog>`:
```
      <Dialog
        triggerButton={triggerButton}
        open={open}
        onOpenChange={setOpen}
        hideClose
        closeButton='Cancel'
      >
```
![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/94d9254b-4259-4eaa-9ae7-8ec105ad547f)

Reply:
I'm not sure this would work. The reason I have cancel and close separate is so I can control there behavior. This is how I'm forcing the copy/view when showing the secrets.

Reply:
I see, at least we could make the styling consistent with other dialogs though? e.g.
![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/ede25991-9013-437d-a0ca-5d7ba7edfe0c)